### ISSUE TITLE
how to train cal with Market-1501 and MSMT17

### ISSUE THREAD

**Issue by JimmyHHua** [guxinqian/Simple-CCReID#5]
Hello! The paper has a   comparison on standard datasets without clotheschanging, could u please tell me how to train the cal with  Market-1501 and MSMT17?

**Comment by guxinqian** [comment]
On standard datasets, CAL will degenerate into NormFace (CosFace/ArcFace with margin=0). You can replace softmax-based cross-entropy loss with cosface/arcface in losses for training on standard datasets

**Comment by JimmyHHua** [comment]
Thank you for your quickly reply, I have sent a email for u ,and wanna communicate more ReID problem with you, thanks again!

**Comment by JimmyHHua** [comment]
> 
你好，博士，我们需要修改的是具体哪一个loss，config里面有如下这些：
`_C.LOSS = CN()
# Classification loss
_C.LOSS.CLA_LOSS = 'crossentropy'
# Clothes classification loss
_C.LOSS.CLOTHES_CLA_LOSS = 'cosface'
# Scale for classification loss
_C.LOSS.CLA_S = 16.
# Margin for classification loss
_C.LOSS.CLA_M = 0.
# Pairwise loss
_C.LOSS.PAIR_LOSS = 'triplet'
# The weight for pairwise loss
_C.LOSS.PAIR_LOSS_WEIGHT = 0.0
# Scale for pairwise loss
_C.LOSS.PAIR_S = 16.
# Margin for pairwise loss
_C.LOSS.PAIR_M = 0.3
# Clothes-based adversarial loss
_C.LOSS.CAL = 'cal'
# Epsilon for clothes-based adversarial loss
_C.LOSS.EPSILON = 0.1
# Momentum for clothes-based adversarial loss with memory bank
_C.LOSS.MOMENTUM = 0.
# -----------------------------------------------------------------------------`
我们需要修改的是LOSS.CLA_LOSS还是LOSS.CLOTHES_CLA_LOSS？

**Comment by JimmyHHua** [comment]
> On standard datasets, CAL will degenerate into NormFace (CosFace/ArcFace with margin=0). You can replace softmax-based cross-entropy loss with cosface/arcface in losses for training on standard datasets

你好，博士，我们需要修改的是具体哪一个loss，config里面有如下这些：
```python
_C.LOSS = CN()
# Classification loss
_C.LOSS.CLA_LOSS = 'crossentropy'
# Clothes classification loss
_C.LOSS.CLOTHES_CLA_LOSS = 'cosface'
# Scale for classification loss
_C.LOSS.CLA_S = 16.
# Margin for classification loss
_C.LOSS.CLA_M = 0.
# Pairwise loss
_C.LOSS.PAIR_LOSS = 'triplet'
# The weight for pairwise loss
_C.LOSS.PAIR_LOSS_WEIGHT = 0.0
# Scale for pairwise loss
_C.LOSS.PAIR_S = 16.
# Margin for pairwise loss
_C.LOSS.PAIR_M = 0.3
# Clothes-based adversarial loss
_C.LOSS.CAL = 'cal'
# Epsilon for clothes-based adversarial loss
_C.LOSS.EPSILON = 0.1
# Momentum for clothes-based adversarial loss with memory bank
_C.LOSS.MOMENTUM = 0.
```
我们想跑market1501，是需要修改那个LOSS，CLA_LOSS 还是CLOTHES_CLA_LOSS？dataloader设置方面需要注意哪些？

**Comment by guxinqian** [comment]
标准数据集可以用我另一个repo跑，https://github.com/guxinqian/Simple-ReID，对应config/res50_cosface_triplet_0.yaml

**Comment by JimmyHHua** [comment]
> 标准数据集可以用我另一个repo跑，[https://github.com/guxinqian/Simple-ReID，对应config/res50_cosface_triplet_0.yaml](https://github.com/guxinqian/Simple-ReID%EF%BC%8C%E5%AF%B9%E5%BA%94config/res50_cosface_triplet_0.yaml)

如果是标准数据的话，是不是就相当于没有criterion_adv损失函数和clothes_classifier这个分支啦

**Comment by guxinqian** [comment]
是的

**Comment by JimmyHHua** [comment]
> 是的

明白了，辛苦解惑~
